parsing_test.go

```diff
@@ -79,3 +79,13 @@ func TestSetURLVars(t *testing.T) {
 	r = SetURLVars(r, vars)
 	assert.Equal(t, vars, GetVars(r))
 }
+
+func TestDecodeVars(t *testing.T) {
+	r, err := http.NewRequest("GET", "/url", nil)
+	assert.NoError(t, err)
```

Comment:
`require.NoError` is preferred in scenarios where the test definitely can't continue if the assumption is wrong. In this case, the test can't go on if `NewRequest` errors.